components/movementsensor/replay/replay.go

```diff
@@ -214,11 +217,18 @@ func (replay *replayMovementSensor) Position(ctx context.Context, extra map[stri
 		return nil, 0, err
 	}
 
-	coordStruct := data.GetFields()["coordinate"].GetStructValue()
+	coordStruct, ok := data.GetFields()["coordinate"]
+	if !ok {
+		return nil, 0, errBadData
+	}
+	altitude, ok := data.GetFields()["altitude_m"]
+	if !ok {
+		return nil, 0, errBadData
+	}
 	return geo.NewPoint(
-			coordStruct.GetFields()["latitude"].GetNumberValue(),
-			coordStruct.GetFields()["longitude"].GetNumberValue()),
-		data.GetFields()["altitude_m"].GetNumberValue(), nil
+			coordStruct.GetStructValue().GetFields()["latitude"].GetNumberValue(),
+			coordStruct.GetStructValue().GetFields()["longitude"].GetNumberValue()),
```

Comment:
did not test if the subfields matched. felt like overkill for this PR but I can if ya want

Reply:
I think since its data from us, being used by us its fine to leave as is.